docs/usage/annotations.rst

```diff
@@ -0,0 +1,118 @@
+* Inputs have to be defined as contiguous memory
+    One :class:`~openqasm3.ast.QubitDeclaration` corresponds to one input
+* Input indices must be selected from a contiguous range of integers starting at `0`
+   No skips, no duplicates
+* Inputs may be declared anywhere in code
+* Input annotations may only appear on a :class:`~openqasm3.ast.QubitDeclaration`
+* Input annotations may only appear once per statement
+
+.. code-block:: openqasm3
+    :linenos:
+
+    // Qubit array
+    @leqo.input <<InputIndex>>
+    qubit[<<length>>] someName;
+
+    // Single qubit
+    @leqo.input <<InputIndex>>
+    qubit someName;
+
+.. _input-memory-layout:
+
+Memory Layout
+~~~~~~~~~~~~~
+
+There is no explicit limit to the size (qubit count) of an input.
+Keep in mind though that the number of physical qubits on a real device is limited.
+See :ref:`reusable-qubit-annotation` for workarounds.
+
+The annotated input size must match the input size of the corresponding node from the `visual model <https://github.com/LEQO-Framework/low-code-modeler>`_.
+
+The backend actively ensures that input memory is initialized.
+All other qubits still have to be assumed to be in an undefined state (See `OpenQasm Specification <https://openqasm.com/language/types.html#qubits#:~:text=Qubits%20are%20initially%20in%20an%20undefined%20state>`_ and :ref:`reusable-qubit-annotation`).
+
+In the future, it will be allowed to input less qubits than specified using the annotation.
+In this case the backend will fill the lowest bytes with the actual input and ensure the upper bytes are initialized to zero:
+
+    .. csv-table:: Example input register of size `7`
+        :header: "0", "1", "2", "3", "4", "5", "6"
+
+        "p[0]", "p[1]", "p[2]", "p[3]", "p[4]", "p[5]", "p[6]"
+        "p[0]", "p[1]", "\|0⟩", "\|0⟩", "\|0⟩", "\|0⟩", "\|0⟩"
+
+.. note::
+    The input might be split into separate non-contiguous memory blocks by the backend.
+
+Output
+------
+
+One output is defines as a single alias (:class:`~openqasm3.ast.AliasStatement`) with a single `@leqo.output` annotation.
+
+* Outputs may be concatenated from multiple non-contiguous blocks of memory.
+* Output indices must be selected from a contiguous range of integers starting at `0`
+    No skips, no duplicates
+* Outputs may be declared anywhere in code
+* Outputs may be used like any other alias
+* Output annotations may only appear above a :class:`~openqasm3.ast.AliasStatement` pointing to qubits
+* Output annotations may only appear once per statement
+
+.. code-block:: openqasm3
+    :linenos:
+
+    @leqo.output <<OutputIndex>>
+    let someOutput = <<Expression>>;
+
+    // Example
+    @leqo.output 0
+    let output1 = a[1:2:3] ++ b[{1,2,3}];
+
+.. note::
+    Your defined alias might never be used in code.
+    Still, you have to define one including an identifier chosen by you.
+    The identifier is insignificant and will be ignored.
+
+.. _reusable-qubit-annotation:
+
+Reusable Qubits
```

Comment:
Maybe add "(Ancilla Qubits)" to the heading (or somewhere in the text). The term "ancilla qubits" is the commonly used term for this kind of workspace qubits and thus should be mentioned here.